lib/handlers/subscribe.js

```diff
@@ -142,13 +142,12 @@ function addSubs (sub, done) {
   const nl = this.nl
   let func = qos > 0 ? client.deliverQoS : client.deliver0
 
-  if (!rap) {
-    const deliverFunc = func
-    func = function handlePacketSubscription (_packet, cb) {
-      _packet = new Packet(_packet, broker)
-      _packet.retain = false
-      deliverFunc(_packet, cb)
-    }
+  const deliverFunc = func
+  func = function handlePacketSubscription (_packet, cb) {
+    _packet = new Packet(_packet, broker)
+    _packet.nl = nl
+    if (!rap) _packet.retain = false
+    deliverFunc(_packet, cb)
```

Comment:
Could you explain this?

Reply:
Originally, the code checked if rap (retain as published) was `false` (which is default) and if so then it wrapped the delivery function and injected / modified the retain flag to `false` to indicate that the message was not retained.

I have made a modification that always wraps the delivery function and always injects nl (nolocal) flag from the subscription into the packet. The `func` is then sent to mqemmiter that uses it whenever a packet is received.

So this way I make the `nl` flag from the subscription accessible in the `deliver0` and `deliverQoS` in the client.js.